packages/rps/src/redux/game-v2/saga.ts

```diff
@@ -0,0 +1,191 @@
+import {take, select, call, put} from 'redux-saga/effects';
+// import { take } from 'redux-saga/effects';
+import {RPSChannelClient} from '../../utils/rps-channel-client';
+import {AppData, hashPreCommit, calculateResult, updateAllocation, Player} from '../../core';
+import {updateChannelState, chooseSalt, resultArrived, restart, FundingSituation} from './actions';
+import {GameState} from './state';
+import {randomHex} from '../../utils/randomHex';
+import {bigNumberify, BigNumber} from 'ethers/utils';
+
+const getGameState = (state: any): GameState => state.game;
+
+export function* gameSaga(channelClient: RPSChannelClient) {
```

Comment:
Would it be clearer to name this object `rpsChannelClient`, to disambiguate from the `ChannelClient` type?